core/bin/external_node/src/main.rs

```diff
@@ -458,6 +593,60 @@ struct Cli {
     /// This is an experimental and incomplete feature; do not use unless you know what you're doing.
     #[arg(long)]
     enable_snapshots_recovery: bool,
+    /// Comma-separated list of components to launch.
+    #[arg(long, default_value = "all")]
+    components: ComponentsToRun,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum Component {
+    HttpApi,
+    WsApi,
+    Tree,
+    TreeApi,
+    Core,
```

Comment:
Yeah currently main node and EN init APIs in two separate places and I am not sure that they can be united. Ideally EN should reuse code for running API and other components from `./core/lib/zksync_core/src/lib.rs`, i will try to make it work but i am not sure that currently it is possible with main node configs vs. EN configs.

Reply:
Running EN/main node in a unified way will be handled by Legolizer, so I think it doesn't make much sense to create some kind of unification logic for the old system. An interim solution would do, IMO.

Reply:
got it - I definitely don't insist making it perfect, especially if it's temporary. 

But it's **really** important for me that we don't keep diverging EN and Main node further. I think this will make the future decentralization harder. Note how the EN and Main Node will converge once we properly integrate the consesnsus.

Seeing the @popzxc 's comment above I'm more optimistic. Will have a closer look at the PR later on.